docs/releasing/publishing-a-pre-release.md

```diff
@@ -0,0 +1,125 @@
+
+Review the docs for [what to do before publishing a release](/docs/releasing/before-publishing-a-release.md) to assess which steps you need to follow for your specific pre-release and ensure you are prepared to publish.
+
+See the [documentation on support branches](https://govuk-design-system-team-docs.netlify.app/how-we-work/version-control/support-branches.html#support-branches) if you need to publish a new pre-release of previous major versions of GOV.UK Frontend.
+
+# Publish a new version of GOV.UK Frontend
+
+Developers should pair on pre-releases. When remote working, it can be useful to be on a call together.
+
+1. Check out the **main** branch and pull the latest changes.
+
+2. Run `nvm use` to make sure you're using the right version of Node.js and npm.
+
+3. Run `npm ci` to make sure you have the exact dependencies installed.
+
+4. Determine the pre-release identifier
+
+   - Use `internal` for internal pre-releases
+   - Use `beta` for beta pre-releases
+
+5. Determine the pre-release version type
+
+   Given the `beta` pre-release identifier:
+
+   - Use `premajor` to bump from `v4.7.0` to `v5.0.0-beta.0`
+   - Use `preminor` to bump from `v4.7.0` to `v4.8.0-beta.0`
+   - Use `prepatch` to bump from `v4.7.0` to `v4.7.1-beta.0`
+
+   Alternatively, when publishing an update to an existing pre-release:
+
+   - Use `prerelease` to bump from `v5.0.0-beta.0` to `v5.0.0-beta.1`
+
+6. Create and check out a new branch (`release-[version-number]`). See the [versioning documentation](/docs/contributing/versioning.md) for more information.
+
+7. If you're publishing a beta pre-release, update the [`CHANGELOG.md`](/CHANGELOG.md) by:
+
+   - changing the 'Unreleased' heading to the new version number and release type. For example, '5.0.0-beta.0 (Pre-release)'
+   - adding a new 'Unreleased' heading above the new version number and release type, so users will know where to add PRs to the changelog
+   - saving your changes
+
+8. Apply the new pre-release version number by running:
+
+   ```shell
+   npm version <PRE-RELEASE TYPE> --preid <PRE-RELEASE IDENTIFIER> --no-git-tag-version --workspace govuk-frontend
+   ```
+
+   This step will update [`govuk-frontend`'s `package.json`](/packages/govuk-frontend/package.json) and project [`package-lock.json`](/package-lock.json) files.
+
+   Do not commit the changes.
+
+9. Update browser data from ["Can I use"](https://caniuse.com) by running:
+
+   ```shell
+   npx update-browserslist-db@latest
+   ```
+
+   This step will update the project [`package-lock.json`](/package-lock.json) file if updates are found.
+
+10. Run `npm run build-release` to:
+
+    - build GOV.UK Frontend into [the package's `/dist`](/packages/govuk-frontend/dist) and [root `/dist`](/dist) directories
+    - commit the changes
+    - push a branch to GitHub
+
+    You will now be prompted to continue or cancel.
+
+11. If you're publishing a beta pre-release, create a pull request
```

Comment:
**note** Figured we don't really need to reverse `internal` pre-releases to `main`, only `beta` ones. Reason for that is that if we alternate `beta` and `internal` pre-releases (for example, to check a bugfix before the next pre-release), that will make the version bump rocky for `beta` release as we'd go from `-beta.X` to `-internal.0` then the next `prerelease` bump would go to `-beta.0` instead of `-beta.X+1`.